### PR TITLE
Feature determine draw

### PR DESCRIPTION
**Pull request by grace000**
https://trello.com/c/7H19qaXy/7-ability-to-determine-draw

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -3,15 +3,15 @@ class Board
...
-        moves.length == 0
+        moves.all? { |move| move.nil? }
     end
 
     def full?
-        moves.length == 9
+        moves.length == 9 && !moves.include?(nil)
     end
 
     def move(token, position)
```
Comment (BWoodfork) [inline lib/board.rb]: Good changes.

File: lib/command_line_input.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-class CommandLineIn
+class CommandLineInput
```
Comment (BWoodfork) [inline lib/command_line_input.rb]: Good name change

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    end
+
+    def play(players, board)
+        players.each { |player| 
+            puts "HELLO PLAYER #{player.token}"
+            puts @presenter.display_board(board.moves)
+            user_coordinate = select_coordinate
+            take_turn(board, user_coordinate, player.token)
+            return if game_over?(board)
+        }
```
Comment (BWoodfork) [inline lib/game.rb]: Usually, when you have an iterator block that uses multiple lines, I'd suggest using the 

`each do` format, as opposed to the block format. The block format is normally used for things that have 1-2 short-ish lines of code. Not a huge deal either way though.

File: lib/game_results.rb [hunk 4]
```diff
@@ -1,45 +1,38 @@
...
+        else 
+            false
+        end
     end
 
-    def draw?(board)
-        board.full?
+    def has_winning_combos?(board)
+        has_winning_row?(board) || has_winning_column?(board)
     end
```
Comment (BWoodfork) [inline lib/game_results.rb]: Good naming

File: lib/game_results.rb [hunk 5]
```diff
@@ -1,45 +1,38 @@
...
-            ]
-        end
+    def has_winning_row?(board)
+        rows = split_board_state(board)
+        rows.any? { |row| return true if all_cells_equal?(row)}
+    end
+
+    def has_winning_column?(board)
+        columns = split_board_state(board)
+        columns.transpose.any? { |column| return true if all_cells_equal?(column)}
```
Comment (BWoodfork) [inline lib/game_results.rb]: Same here. Don't need to explicitly return true.

File: lib/game_results.rb [hunk 6]
```diff
@@ -1,45 +1,38 @@
...
+    end
+
+    def split_board_state(board)
+        board.moves.each_slice(3).to_a
```
Comment (BWoodfork) [inline lib/game_results.rb]: Why are you converting to an array here? It's not already an array?

File: lib/setup.rb [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+    def assign_player_token(input)
+        puts "Thanks for selecting #{input}."
+        player = Player.new(input)
+        @players << player
+    end
+
+    def start_game_engine
+        puts Prompt::WELCOME
+        2.times { |i|
```
Comment (BWoodfork) [inline lib/setup.rb]: What does `i` represent here?

File: lib/game.rb [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+    def take_turn(board, position, token)
+        input_validator = InputValidator.new
+        if input_validator.valid_coordinate?(board.moves, position)
+            board.move(token, position)
+        else
+            user_coordinate = select_coordinate
+            take_turn(board, user_coordinate, token)
+        end
+    end
```
Comment (BWoodfork) [inline lib/game.rb]: Are there tests for this method?

File: lib/game_results.rb (hunk 6 above)
Comment (grace000) [reply lib/game_results.rb]: I'm converting to a 2D array here so that I can have rows and then transpose for columns. [X,X,X,X,X,X] --> [[X,X,X],[X,X,X]]

**Pull request merged**